src/app/story/directives/basic.component.ts

```diff
@@ -100,11 +114,72 @@ export class BasicComponent implements OnDestroy, DoCheck {
     return (this.story && this.story.publishedAt) ? true : false;
   }
 
+  get undeletedVersions(): AudioVersionModel[] {
+    if (this.story && this.story.versions) {
+      return this.story.versions.filter(v => !v.isDestroy);
+    } else {
+      return null;
+    }
+  }
+
+  get versionsChanged(): boolean {
+    return this.story && this.story.changed('versions', true);
+  }
+
+  get versionsInvalid(): boolean {
+    return this.story && !!this.story.invalid('versions', this.strict);
+  }
+
+  updateVersions(templateIds: number[]) {
+    let used = {};
+    this.story.versions.forEach(v => {
+      if (v.template) {
+        used[v.template.id] = true;
+        if (templateIds.indexOf(v.template.id) > -1) {
+          v.isDestroy = false;
+        } else if (v.isNew) {
+          this.story.removeRelated(v);
+        } else {
+          v.isDestroy = true;
+        }
+      }
+    });
+    templateIds.forEach(id => {
+      if (!used[id]) {
+        this.story.versions.push(new AudioVersionModel({
+          series: this.story.parent,
+          template: this.versionTemplates[id]
+        }));
+      }
+    });
+  }
+
   toggleShowReleaseAt() {
     this.showReleasedAt = !this.showReleasedAt;
     if (this.story.releasedAt) {
       this.story.releasedAt = null;
     }
   }
 
+  loadVersionTemplates() {
+    this.story.getSeriesTemplates().subscribe(tdocs => {
```

Comment:
Navigating from the series episode list to a story threw an error `this.story.getSeriesTemplates is not a function` and would be worth checking here

Reply:
Yeah, this is the error with the TabService.  The first/initial time this is called, it gets a SeriesModel, which causes all sorts of trouble.

I'll throw in some patch fixes to make sure `story instanceof StoryModel` ... then remove those when we actually fix the styleguide.

Reply:
Not sure I'm on the same page with the fix that's needed in the styleguide. TabService isn't very smart, just a model Observable.

Reply:
Update: fixed by providing the TabService at the component level, instead of the shared-module.